lib/resty/core/time.lua

```diff
@@ -26,6 +41,81 @@ function ngx.time()
 end
 
 
+function ngx.update_time()
+    C.ngx_http_lua_ffi_update_time()
+end
+
+
+function ngx.today()
+    -- the format of today is 2010-11-19
+    local today_buf_size = 10
+    local buf = get_string_buf(today_buf_size)
```

Comment:
if the buffer size is less than `4096`, this buffer can be placed to top level.

https://github.com/spacewander/lua-resty-core/blob/9e4147e8a311a781a478e57fda0c547e9fe5bb82/lib/resty/core/base.lua#L182

Reply:
@membphis 
I am afraid this change will make code hard to comprehend. All those top-level buffer actually point to one buffer, which is counter-intuitive. And function call is cheap.

Reply:
@spacewander We should reuse the `resty.core.base` buffer instead of creating an extra cdata buffer. The former is readily usable at this point anyway.

Reply:
@agentzh I means we can cache the `resty.core.base` buffer at top-level, then we can use it directly to avoid an extra function call.

```lua
local base = require "resty.core.base"
local buf = base.get_string_buf(4096)

-- some code

function ngx.today()
    -- the format of today is 2010-11-19
    local today_buf_size = 10
    C.ngx_http_lua_ffi_today(buf)
    return ffi_str(buf, today_buf_size)
end

```

Reply:
@membphis Your way can be more expensive since the size of the string buffer of `resty.core.base` is tunable, which means it is not necessarily 4096. So you may still allocate a separate memory buffer than necessary when the actual buffer size is smaller than 4096 while the `today()` function only needs 10 bytes, for example.

I think this is a case of over-optimization. The LuaJIT compiler is usually smart enough to inline the Lua function call and optimize the linear code paths reducing branch mispredictions.